common/lib/plugins/strategy/fastest_response/fastest_response_strategy_plugin.ts

```diff
@@ -0,0 +1,132 @@
+import { HostChangeOptions } from "../../../host_change_options";
+import { RandomHostSelector } from "../../../random_host_selector";
+import { Messages } from "../../../utils/messages";
+import { logAndThrowError } from "../../../utils/utils";
+
+export class FastestResponseStrategyPlugin extends AbstractConnectionPlugin {
+  static readonly FASTEST_RESPONSE_STRATEGY_NAME: string = "fastestResponse";
+  private static readonly subscribedMethods = new Set<string>(["notifyHostListChanged", "acceptsStrategy", "getHostInfoByStrategy"]);
+  protected static readonly cachedFastestResponseHostByRole: CacheMap<string, HostInfo> = new CacheMap<string, HostInfo>();
+  protected cacheExpirationNanos: bigint;
+  protected hostResponseTimeService: HostResponseTimeService;
+  protected readonly properties: Map<string, any>;
+  private pluginService: PluginService;
+  private randomHostSelector: RandomHostSelector = new RandomHostSelector();
+
+  constructor(pluginService: PluginService, properties: Map<string, any>, hostResponseTimeService?: HostResponseTimeService) {
+    super();
+    this.pluginService = pluginService;
+    this.properties = properties;
+    this.hostResponseTimeService =
+      hostResponseTimeService ??
+      new HostResponseTimeServiceImpl(pluginService, properties, WrapperProperties.RESPONSE_MEASUREMENT_INTERVAL_MILLIS.get(this.properties));
+    this.cacheExpirationNanos = BigInt(WrapperProperties.RESPONSE_MEASUREMENT_INTERVAL_MILLIS.get(this.properties) * 1_000_000);
+  }
+
+  public getSubscribedMethods(): Set<string> {
+    return FastestResponseStrategyPlugin.subscribedMethods;
+  }
+
+  async connect(
+    hostInfo: HostInfo,
+    props: Map<string, any>,
+    isInitialConnection: boolean,
+    connectFunc: () => Promise<ClientWrapper>
+  ): Promise<ClientWrapper> {
+    const result = await connectFunc();
+    if (isInitialConnection) {
+      this.hostResponseTimeService.setHosts(this.pluginService.getHosts());
+    }
+    return result;
+  }
+
+  async forceConnect(
+    hostInfo: HostInfo,
+    props: Map<string, any>,
+    isInitialConnection: boolean,
+    forceConnectFunc: () => Promise<ClientWrapper>
+  ): Promise<ClientWrapper> {
+    const result = await forceConnectFunc();
+    if (isInitialConnection) {
+      this.hostResponseTimeService.setHosts(this.pluginService.getHosts());
+    }
+    return result;
+  }
+
+  acceptsStrategy(role: HostRole, strategy: string) {
+    return FastestResponseStrategyPlugin.FASTEST_RESPONSE_STRATEGY_NAME === strategy;
+  }
+
+  getHostInfoByStrategy(role: HostRole, strategy: string, hosts?: HostInfo[]): HostInfo | undefined {
+    if (!this.acceptsStrategy(role, strategy)) {
+      logAndThrowError(Messages.get("FastestResponseStrategyPlugin.unsupportedHostSelectorStrategy", strategy));
+    }
+    // The cache holds a host with the fastest response time.
+    // If the cache doesn't have a host for a role, it's necessary to find the fastest node in the topology.
```

Comment:
Small nit, can we please use `host` or `instance` instead of `node` for consistency